builtin/plugins/chainconfig/chainconfig.go

```diff
@@ -350,6 +366,93 @@ func EnableFeatures(ctx contract.Context, blockHeight, buildNumber uint64) ([]*F
 	return enabledFeatures, nil
 }
 
+// GetPendingActions returns a list of actions that are supported by majority of validators
+func GetPendingActions(ctx contract.Context, buildNumber uint64) ([]*Action, error) {
```

Comment:
Can rename this to `HarvestPendingActions` and delete the action from the context right after `actions = append(actions, &action)`. That way you don't need to mess with the contract state in `ChainCconfigManager.UpdateConfig()`.

Reply:
Done